index.js

```diff
@@ -1,6 +1,41 @@
-/* jshint node: true */
 'use strict';
 
 module.exports = {
-  name: 'ember-legacy-controllers'
+  name: require('./package').name,
+
+  init() {
+    this._super.init && this._super.init.apply(this, arguments);
+    this.hasLegacyViewSupport = this.project.config(process.env.EMBER_ENV).EmberENV._ENABLE_LEGACY_VIEW_SUPPORT;
```

Comment:
I think you'll need to move this down into `included`, `process.env.EMBER_ENV` is only set when doing a build and therefore running this in `init` will be doing `this.project.config(undefined)` for any non-build commands (e.g. `ember try:*`).